SoulTicket internal implementation

Arcaratus opened the pull request:
-Removes all usage of deprecated methods in favor of the new Soul Ticket methods
-Also fixes toggleable sigils

Arcaratus committed:
Fix the Blood Tank BB

Arcaratus committed:
Merge branch '1.12' of https://github.com/WayofTime/BloodMagic into 1.12

Arcaratus committed:
Merge branch '1.12' of https://github.com/WayofTime/BloodMagic into 1.12

Arcaratus committed:
Merge branch '1.12' of https://github.com/WayofTime/BloodMagic into 1.12

Arcaratus committed:
Add modid to command localizations to prevent conflicts

Arcaratus committed:
Fixed the items not being drawn on the right Y-level for the Sigil of Holding HUD
Corrected localizations of other lang files

Arcaratus committed:
Merge branch '1.12' of https://github.com/WayofTime/BloodMagic into 1.12

Arcaratus committed:
SoulTicket internal implementation

TehNut commented on src/main/java/WayofTime/bloodmagic/core/data/SoulNetwork.java:
```diff
@@ -93,30 +104,34 @@ public int syphon(int amount) {
     }
 
-    public BooleanResult<Integer> syphonAndDamage(EntityPlayer user, SoulTicket ticket) {
+    public boolean syphonAndDamage(EntityPlayer user, SoulTicket ticket) {
```
...why?

TehNut commented on src/main/java/WayofTime/bloodmagic/core/data/SoulTicket.java:
```diff
@@ -30,4 +35,45 @@ public ITextComponent getDescription() {
+    /**
+     * @return A description in the format block|dimensionID|pos
+     */
+    public static SoulTicket BLOCK(World world, BlockPos pos, int amount) {
```
Method names should not be capital

Arcaratus committed:
do what TehNut says

TehNut commented on src/main/java/WayofTime/bloodmagic/core/data/SoulTicket.java:
```diff
@@ -30,4 +35,45 @@ public ITextComponent getDescription() {
+
+    // TODO maybe make it check the amount??
+    @Override
+    public boolean equals(Object o) {
```
If you override `equals`, you have to override `hashCode` as well.

Arcaratus committed:
implement hashCode()

Arcaratus committed:
Fix toggleable sigils draining on r-click when it shouldn't
Also moved the ItemSigil and ItemSigilToggleable to the sigil package (why wasn't it there???)
